numba/typed/typedlist.py

```diff
@@ -235,6 +235,10 @@ def _typed(self):
         """
         return self._list_type is not None
 
+    @property
+    def _dtype(self):
+        return self._list_type.dtype if self._typed else None
```

Comment:
@esc does it make sense to get a value (`None`) back from a this if it's untyped?

Reply:
The only other alternative I see is to raise an exception, and I am not sure that is much better.

Reply:
This doesn't have an equivalent within an `@njit` function, since the list must be typed for the function to compile. You could convincingly however check the `_dtype` before refinement and I do wonder what that would result in.

Reply:
If a method is being added to this class and the implementer sees the `_dtype` and thinks "I'll query that" is there a chance that a `None` return would be misconstrued as the dtype/cause other confusion? In which case, probably better to just raise? I think this would also be consistent with https://github.com/numba/numba/blob/f4e31765755f25c582987c1b4abd5680a9eba5a4/numba/typed/typedlist.py#L228-L229 ?

Reply:
@stuartarchibald I am sold, let's raise. @rohitsanj can you make the requested change?